main.go

```diff
@@ -82,14 +82,37 @@ func main() {
 		log.Fatalf(usage)
 	}
 
-	url, err := url.Parse(args[0])
+	rawURL := normalizeURL(args[0])
+	url, err := url.Parse(rawURL)
 	if err != nil {
 		log.Fatalf("could not parse url %q: %v", args[0], err)
 	}
 
 	visit(url)
 }
 
+// normalizeURL normalizes the url input.
+// If the scheme is missing, we assume it's "https" (or "http" if port 80).
+// If the host is missing, we assume it's "localhost".
+func normalizeURL(rawURL string) string {
+	if strings.Index(rawURL, "://") > -1 {
+		return rawURL
+	}
+
+	hostport := strings.Split(rawURL, ":")
+	if len(hostport) == 2 {
+		if hostport[0] == "" {
+			rawURL = "localhost:" + hostport[1]
+		}
+
+		if strings.Index(hostport[1], "80") == 0 {
+			return "http://" + rawURL
+		}
+	}
```

Comment:
There's an interesting feature of `url.Parse` in that it doesn't populate the `url.Host` field if there is no Scheme specified. I just learned this about 2 minutes ago. 😒 

If we pass in `www.google.com` to `url.Parse()`, it doesn't make any assumptions about the scheme. We need to do some processing before trying to use `url.Parse()`, to prefix the URL with a scheme (if needed).